common/util/sha256_test.cc

```diff
@@ -0,0 +1,116 @@
+// Copyright 2017-2023 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "common/util/sha256.h"
+
+#include <array>
+#include <cstdio>
+#include <string>
+#include <vector>
+
+#include "absl/strings/escaping.h"
+#include "absl/strings/string_view.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace verible {
+namespace {
+
+using ::testing::Eq;
+
+// The tests are comparing this SHA256 implementation and the SHA256 digest
+// produced by the OpenSSL SHA256 implementation. The following code was used to
+// create the expected SHA256 digests:
+//
+// #include "openssl/sha.h"
+//
+// std::string SslSHA256Digest(absl::string_view content) {
+//   std::array<unsigned char, SHA256_DIGEST_LENGTH> buf;
+//   ::SHA256(reinterpret_cast<const unsigned char *>(content.data()),
+//            content.size(), buf.data());
+//   return absl::BytesToHexString(absl::string_view(
+//       reinterpret_cast<const char *>(buf.data()), buf.size()));
+// }
+//
+// (OpenSSL is not linked to avoid a dependency bloat)
+// Alternative: `printf "banana" | sha256sum`
+
+TEST(Sha256, DigestsAreEqual) {
+  const std::string kOpenSslSha256BananaDigest =
+      "b493d48364afe44d11c0165cf470a4164d1e2609911ef998be868d46ade3de4e";
+  EXPECT_THAT(Sha256Hex("banana"), Eq(kOpenSslSha256BananaDigest));
+}
+
+TEST(Sha256, NonAsciiDigestsAreEqual) {
+  const std::string kOpenSslSha256JaBananaDigest =
+      "787bcc7042939ad9607bc8ca87332e4178716be0f0b890cbf673884d39d8ff79";
+  EXPECT_THAT(Sha256Hex("バナナ"), Eq(kOpenSslSha256JaBananaDigest));
+}
+
+TEST(Sha256, LargeInputDigestsAreEqual) {
+  const std::string kLargeText = R"(
```

Comment:
I'd make that a `static constexpr absl::string_view` (similar with the other `std::string` constants in this test)